### PR TITLE
add ami to initialisms

### PR DESCRIPTION
**Pull request by brycahta**
Issue #, if available: N/A

Description of changes:
* Add `ami` entry to initialisms with test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by RedbackThomson — APPROVED** [review]
/lgtm

File: pkg/names/names.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ var (
 		// Prevent "MultipartUpload" from becoming "MultIPartUpload"
 		{"Ip", "IP", "ip", regexp.MustCompile("Ip(?!art)", regexp.None)},
 		// Easy find-and-replacements...
+		{"Ami", "AMI", "ami", nil},
```
Comment (jaypipes) [inline pkg/names/names.go:71]: @brycahta @RedbackThomson I think this is going to have issues with "Family"... which this is going to turn into "FAMIly"

Comment (A-Hilaly) [reply pkg/names/names.go:71]: We can use something similar to what we used for URI/SecURIty: `{"Uri", "URI", "uri", regexp.MustCompile("(?!sec)uri(?!ty)|(Uri)", regexp.None)},`

Comment (brycahta) [reply pkg/names/names.go:71]: Good points, I was surprised no tests were failing tbh. Let me submit a revision to handle "Family"

Comment (brycahta) [reply pkg/names/names.go:71]: https://github.com/aws-controllers-k8s/code-generator/pull/167

**Pull request merged**